gufe/components/explicitmoleculecomponent.py

```diff
@@ -82,15 +83,18 @@ def _check_partial_charges(mol: RDKitMol) -> None:
                 raise ValueError(errmsg)
 
     if np.all(np.isclose(p_chgs, 0.0)):
-        wmsg = f"Partial charges provided all equal to " "zero. These may be ignored by some Protocols."
+        wmsg = "Partial charges provided all equal to zero. These may be ignored by some Protocols."
         warnings.warn(wmsg)
     else:
-        wmsg = (
+        message = (
             "Partial charges have been provided, these will "
             "preferentially be used instead of generating new "
-            "partial charges"
+            "partial charges."
         )
-        warnings.warn(wmsg)
+        if logger is None:
+            logger = logging.getLogger()
```

Comment:
```suggestion
            logger = logging.getLogger(__name__)
```

Reply:
This will scope this logging message to a logger corresponding to this module, instead of just the root logger.